### PR TITLE
screen-locker: Make xautolock optional, reorganize options

### LINKED ISSUE #2341: services.screen-locker: Option to disable xautolock
### Description

And maybe default to disabled?

In my experience on other distros, `xss-lock` has been sufficient for all of my screen locking needs. Since it receives the signal from X that the user is idle, it seems to accomplish the same time-based locking that xautolock does.

I've noticed since switching to home-manager for my user configs that my screen locks while I'm doing things like watching a video or on a call, and I suspect that it's because xautolock is a bit more aggressive and/or disregards the screensaver inhibition during said activities.

### PR DESCRIPTION
**Pull request by jrobsonchase**
### Description

xautolock isn't really needed to trigger xss-lock on the basis of time
since the built-in screensaver functionality of X serves as one of the
event sources for xss-lock. Keeping it around and defaulting to
"enabled" to avoid unexpected breakage.

Also shuffled around the options to submodules for xss-lock and
xautolock to get rid of prefixes in option names and to make detectSleep
a bit clearer.

Resolves #2341 

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

### CODE REVIEW

**Comment by jrobsonchase** [comment]
Hmm, I can't actually figure out how to make the expected warnings not specific to my file paths.

Regexes like I have right now don't work since it uses `assertFileContent`, and paths (i.e. `${./moved-options.nix}`) point to the nix store rather than the location in the warning.

I don't see any existing examples in the home-manager tests to go off of. What's the move here?

**Comment by jrobsonchase** [comment]
OK, finally figured out how to get the set of files that set an option, so the warning test is now working.

**Review by thiagokokada — APPROVED** [review]
LGTM.

File: modules/services/screen-locker.nix
Comment (thiagokokada) [inline modules/services/screen-locker.nix:82]: @jrobsonchase can you add yourself as a maintainer?

Also, add yourself to `CODEOWNERS` file.

**Comment by thiagokokada** [comment]
@berbiche @sumnerevans @rycee If either of you could also review this PR. I am quite interested on this being merged (I was planning to do something similar).

File: modules/services/screen-locker.nix
Comment (berbiche) [inline modules/services/screen-locker.nix:10]:
```suggestion
  meta.maintainers = [ lib.hm.maintainers.jrobsonchase ];
```

File: .github/CODEOWNERS
Comment (berbiche) [inline .github/CODEOWNERS:367]: While there is no precise ordering in the file, could you move this line to preserve some of the ordering?

File: modules/services/screen-locker.nix
Comment (berbiche) [inline modules/services/screen-locker.nix:125]: Does `$XDG_SESSION_ID` correctly get interpolated in the service script execution?

File: tests/modules/services/screen-locker/basic-configuration.nix
Comment (berbiche) [inline tests/modules/services/screen-locker/basic-configuration.nix:15]:
```suggestion
    services.screen-locker = {
      enable = true;
      inactiveInterval = 5;
      lockCmd = "${pkgs.i3lock}/bin/i3lock -n -c AA0000";
      xss-lock = { extraOptions = [ "-test" ]; };
      xautolock = {
        enable = true;
        detectSleep = true;
        extraOptions = [ "-test" ];
      };
    };
    
    test.stubs.i3lock = { };
    test.stubs.xss-lock = { };
```

We need a stubbed package for i3lock because we don't want to pull the package when running the test suite.

Can you also make this change in the other tests?

File: modules/services/screen-locker.nix
Comment (berbiche) [inline modules/services/screen-locker.nix:102]: Can you make the xss-lock package used configurable through `services.screen-locker.xss-lock.package`?

File: modules/services/screen-locker.nix
Comment (berbiche) [inline modules/services/screen-locker.nix:123]: Same thing for the configurable package if possible

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

**Comment by jrobsonchase** [comment]
@berbiche Comments should all be addressed and ready for another look

**Review by berbiche — APPROVED** [review]
Thanks.

This is great :+1: 
I'll wait for the checks to pass before merging.

**Pull request merged**
